### PR TITLE
Replace `ArgumentDefinition` with a refactored `StrawberryArgument`

### PR DESCRIPTION
**Pull request by patrick91**
Continuation of https://github.com/strawberry-graphql/strawberry/pull/752

This PR renames `ArgumentDefinition` to `StrawberryArgument`

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Internal refactory. Renamed ArgumentDefition to StrawberryArgument

File: strawberry/arguments.py [hunk 1]
```diff
@@ -137,22 +207,23 @@ def convert_arguments(
...
     return kwargs
 
 
-def argument(description: Optional[str] = None) -> StrawberryArgument:
-    return StrawberryArgument(description=description)
+def argument(description: Optional[str] = None) -> StrawberryArgumentAnnotation:
+    return StrawberryArgumentAnnotation(description=description)
 
 
+# TODO: check exports
```
Comment (BryceBeagle) [inline strawberry/arguments.py:224]: We might want to create an issue to just look at imports/exports project wide and make sure they are consistent

Comment (patrick91) [reply strawberry/arguments.py:224]: Done :) https://github.com/strawberry-graphql/strawberry/issues/806

File: tests/utils/test_arguments_converter.py [hunk 2]
```diff
@@ -2,17 +2,17 @@
...
 import strawberry
-from strawberry.arguments import UNSET, ArgumentDefinition, convert_arguments
+from strawberry.arguments import UNSET, StrawberryArgument, convert_arguments
 
 
 def test_simple_types():
     args = {"integer": 1, "string": "abc", "float": 1.2, "bool": True}
 
     arguments = [
-        ArgumentDefinition(name="integer", type=int, origin_name="integer"),
```
Comment (BryceBeagle) [inline tests/utils/test_arguments_converter.py:12]: Is it necessary for these tests to specify _both_ `python_name` and `graphql_name`?

File: strawberry/arguments.py [hunk 3]
```diff
@@ -1,25 +1,110 @@
...
+        self.description = description
+        self.default_value = default_value
+
+    @property
+    def graphql_name(self) -> Optional[str]:
+        if self._graphql_name:
+            return self._graphql_name
+        if self.python_name:
+            return to_camel_case(self.python_name)
+        return None
```
Comment (BryceBeagle) [inline strawberry/arguments.py:59]: Should add a test for coverage here

**Review by BryceBeagle — CHANGES_REQUESTED** [review]

File: tests/utils/test_arguments_converter.py (hunk 2 above)
Comment (patrick91) [reply tests/utils/test_arguments_converter.py:12]: updated to pass graphql_name=None :)

File: strawberry/arguments.py (hunk 3 above)
Comment (patrick91) [reply strawberry/arguments.py:59]: I'd say let's skip it for now, as this will change when we have StrawberryList :)

Comment (BryceBeagle) [reply strawberry/arguments.py:59]: Why would `StrawberryList` change the name calculation?

Comment (patrick91) [reply strawberry/arguments.py:59]: we only have python_name equal to None when using lists (children of lists have no python name nor graphql name)

Comment (BryceBeagle) [reply strawberry/arguments.py:59]: Oh right :slightly_smiling_face:

**Review by BryceBeagle — APPROVED** [review]

**Pull request merged**